BrainPortal/app/controllers/portal_controller.rb

```diff
@@ -46,6 +46,15 @@ def welcome #:nodoc:
     @default_data_provider  = DataProvider.find_by_id(current_user.meta["pref_data_provider_id"])
     @default_bourreau       = Bourreau.find_by_id(current_user.meta["pref_bourreau_id"])
 
+    @tool_count =  Tool
```

Comment:
This expression returns a completely different number of tools compared to `current_user.available_tools` (which is used in the page to generate the table). In fact, right now my page says "out of 50" and yet the table lists 161 tools.

I think this should be changed to "@tool_names =" to generate a list of names, not just a count, and then the array_to_table in the page should use that array too.